src/Compilers/CSharp/Test/Semantic/Semantics/PropertyFieldKeywordTests.cs

```diff
@@ -1350,9 +1388,61 @@ public S(int arg)
 ");
             var accessorBindingData = new SourcePropertySymbolBase.AccessorBindingData();
             comp.TestOnlyCompilationData = accessorBindingData;
+            Assert.NotEmpty(comp.GetTypeByMetadataName("S").GetMembers(GeneratedNames.MakeBackingFieldName("P")));
+            comp.VerifyDiagnostics();
+
+            Assert.Equal(0, accessorBindingData.NumberOfPerformedAccessorBinding);
+        }
 
+        [Fact]
+        public void SpeculativeSemanticModelTest()
+        {
+            var comp = CreateCompilation(@"
+class C
+{
+    public int P { get => 0; }
+}
+");
+            var accessorBindingData = new SourcePropertySymbolBase.AccessorBindingData();
+            comp.TestOnlyCompilationData = accessorBindingData;
+            Assert.Empty(comp.GetTypeByMetadataName("C").GetMembers(GeneratedNames.MakeBackingFieldName("P")));
             comp.VerifyDiagnostics();
 
+            var tree = comp.SyntaxTrees[0];
+            var token = tree.GetRoot().DescendantTokens().Single(t => t.IsKind(SyntaxKind.NumericLiteralToken));
+
+            var model = comp.GetSemanticModel(comp.SyntaxTrees[0]);
+            var identifier = SyntaxFactory.IdentifierName(SyntaxFactory.Identifier(leading: default, contextualKind: SyntaxKind.FieldKeyword, text: "field", valueText: "field", trailing: default));
+            model.TryGetSpeculativeSemanticModel(token.SpanStart, identifier, out var speculativeModel);
+            Assert.Empty(comp.GetTypeByMetadataName("C").GetFieldsToEmit());
```

Comment:
>Assert.Empty(comp.GetTypeByMetadataName("C").GetFieldsToEmit()); [](http://example.com/codeflow?start=12&length=64)

Are you following some specific suggestion by adding this assert? Otherwise, I would remove it because it is going to trigger some work and it is not clear that we want it to be done at this point.

Reply:
Or is it your intent to call it before and call it after the speculation in the next test? Just want to make sure I understand your intent.

Reply:
@AlekseyTs My intent for the two tests was:

- Let one test bind the original tree first (`SpeculativeSemanticModelTest`) then see if there are emitted fields if we speculated with `field`. This was following one of your suggestion in a previous PR, which I recorded as a PROTOTYPE that I should delete in this PR:

    > Speculating with a field keyword within an accessor of a regular property then emitting shouldn't emit the backing field.
    
- Let the other test bind the speculated tree first (`SpeculativeSemanticModelTest2`).

    Since the backing field wasn't marked as calculated, this caused `GetFieldsToEmit()` to return a backing field, which I'm not sure is intended.

Meaning that, currently, the order of binding changes `GetFieldsToEmit()` behavior